core/src/main/java/io/questdb/cairo/mv/DefinitionFileWriter.java

```diff
@@ -0,0 +1,428 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2024 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.cairo.mv;
+
+import io.questdb.cairo.CairoConfiguration;
+import io.questdb.cairo.VarcharTypeDriver;
+import io.questdb.cairo.vm.Vm;
+import io.questdb.cairo.vm.api.MemoryCARW;
+import io.questdb.cairo.vm.api.MemoryCMARW;
+import io.questdb.std.BinarySequence;
+import io.questdb.std.FilesFacade;
+import io.questdb.std.Long256;
+import io.questdb.std.MemoryTag;
+import io.questdb.std.Misc;
+import io.questdb.std.Mutable;
+import io.questdb.std.Transient;
+import io.questdb.std.Unsafe;
+import io.questdb.std.Vect;
+import io.questdb.std.str.LPSZ;
+import io.questdb.std.str.Utf8Sequence;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.Closeable;
+import java.io.IOException;
+
+import static io.questdb.cairo.mv.DefinitionFileUtils.*;
+
+public class DefinitionFileWriter implements Closeable, Mutable {
+    private final BlockMemoryHandleImpl blockMemoryHandle = new BlockMemoryHandleImpl();
+    private final FilesFacade ff;
+    private int blockCount;
+    private long blockOffset;
+    private MemoryCMARW file;
+    private boolean isCommitted;
+    private MemoryCARW memory;
+
+    public DefinitionFileWriter(FilesFacade ff) {
+        this.ff = ff;
+    }
+
+    public AppendOnlyBlock append() {
+        return blockMemoryHandle.reset(0);
+    }
+
+    @Override
+    public void clear() {
+        Misc.free(memory);
+        Misc.free(file);
+        isCommitted = false;
+        blockOffset = 0;
+        blockCount = 0;
+    }
+
+    @Override
+    public void close() throws IOException {
+        clear();
+        memory = null;
+        file = null;
+    }
+
+    public boolean commit() {
+        if (isCommitted) {
+            return false;
+        }
+
+        // exclude checksum but include block count
```

Comment:
nit: this comment looks obsolete considering that both checksum and block counts are written on L95-96

Reply:
addressed